src/test/java/io/vertx/ext/amqp/SenderWithBrokerTest.java

```diff
@@ -202,6 +204,76 @@ public void testSendBasicMessage(TestContext context) {
     });
 
 
+    asyncRecvMsg.awaitSuccess();
+
+  }
+
+  @Test(timeout = 20000)
+  @Repeat(10)
+  public void testDynamicSenderWithOptions(TestContext context) {
+    String address = UUID.randomUUID().toString();
+    String sentContent = "myMessageContent-" + address;
+
+    Async asyncRecvMsg = context.async();
+    AtomicBoolean opened = new AtomicBoolean();
+
+    client = AmqpClient.create(vertx,
+      new AmqpClientOptions().setHost(host).setPort(port).setUsername(username).setPassword(password));
+
+    AtomicReference<AmqpSender> reference = new AtomicReference<>();
+    AtomicReference<Context> exec = new AtomicReference<>();
+    client.connect(res -> {
+      context.assertTrue(res.succeeded());
+
+      res.result().createSender(null,
+        new AmqpSenderOptions()
+        .setDynamic(true)
+        .setLinkName("my-link")
+        .setAutoDrained(false)
+        .setAutoSettle(false),
+        sender -> {
+        context.assertTrue(sender.succeeded());
+        context.assertNotNull(sender.result().address());
+        reference.set(sender.result());
+        exec.set(Vertx.currentContext());
+      });
+    });
+
+    await().until(() -> exec.get() != null);
+
+    ProtonClient proton = ProtonClient.create(vertx);
```

Comment:
I think using ProtonServer rather than the broker+client would be a far better test here, so you can actually ensure a specific dynamic address allocated is returned from the address call. E.g. the equivalent functionality test from vertx-proton: https://github.com/vert-x3/vertx-proton/blob/3.6.3/src/test/java/io/vertx/proton/ProtonClientTest.java#L403